nixos/modules/services/misc/sourcehut/default.nix

```diff
@@ -0,0 +1,168 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sourcehut;
+  cfgIni = cfg.settings;
+
+  # Specialized python containing all the modules
+  python = pkgs.sourcehut.python.withPackages (ps: with ps; [
+    gunicorn
+    # Sourcehut services
+    buildsrht dispatchsrht gitsrht hgsrht listssrht mansrht
+    metasrht pastesrht todosrht
+  ]);
+in {
+  imports =
+    [
+      ./git.nix
+      ./hg.nix
+      ./todo.nix
+      ./man.nix
+      ./meta.nix
+      ./paste.nix
+    ];
+
+  options.services.sourcehut = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable sourcehut - git hosting, continuous integration, mailing list, ticket tracking,
+        task dispatching, wiki and account management services.
+      '';
+    };
+
+    services = mkOption {
+      type = types.nonEmptyListOf (types.enum [ "builds" "dispatch" "git" "hg" "lists" "man" "meta" "paste" "todo" ]);
+      default = [ "builds" "dispatch" "git" "hg" "lists" "man" "meta" "paste" "todo" ];
+      description = ''
+        Services to enable on the sourcehut network.
+      '';
+    };
+
+    address = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+      '';
+    };
+
+    python = mkOption {
+      internal = true;
+      type = types.package;
+      default = python;
+      description = ''
+        The python package to use. It should contain references to the *srht modules and also
+        gunicorn.
+      '';
+    };
+
+    statePath = mkOption {
+      type = types.path;
+      default = "/var/sourcehut";
+      description = ''
+        Root state path for the sourcehut network.
+      '';
+    };
+
+    settings = mkOption {
+      type = with types; attrsOf (attrsOf (nullOr (either bool (either int (either float (either str path))))));
```

Comment:
This will produce a crazy type like the one in #86402. Can you add a decription with `// { description = "..."; }`?

Reply:
Isn't there already a description for this in line 73?

Reply:
I think he meant an example maybe?